### DISCUSSION TITLE
Prevent linkify working on URLs already inside <a> tags?

### DISCUSSION THREAD

**Discussion by raybellis** [yuin/goldmark#231] [Ideas]
Our Hugo site content (using unsafe mode) has a bunch of links where the text contents are full URLs that are already enclosed inside `<a>` tags, e.g:

    <a href="/test"> https://example.com/test/ </a>

With linkify enabled, that inner URL gets wrapped in another set of `<a>` tags, but since `<a>` tags can't legally nest, the resulting DOM elements are:

     <a href="/test"></a>
     <a href="https://example.com/test/"> https://example.com/test/ </a>

The clickable URL text ends up as a link to the fully qualified URL from the text, instead of the relative URL from the original `href` attribute.

(Yes, I know we could use normal Markdown syntax for links but we get more control over the `rel` and `target` attributes of the `<a>` tag like this).

Would it be feasible for linkify to detect such links and _not_ wrap them a second time?

**Comment by kpm** [comment]
I just posted a related feature idea, which would allow more control over attributes, which would allow using safe mode:  https://github.com/yuin/goldmark/discussions/370
